jupyterlab_gitsync/src/components/toolbar_buttons.tsx

```diff
@@ -0,0 +1,169 @@
+import * as React from 'react';
+import { classes, style } from 'typestyle';
+import {
+  toolbarItemClass,
+  toolbarButtonClass,
+  toolbarButtonIconClass,
+} from '../style/toolbar';
+
+import { Props } from './panel';
+
+export interface State {
+  state: string;
+  label: string;
+  style: ReturnType<typeof style>;
+}
+
+class ToolbarButton extends React.Component<Props, State> {
+  constructor(
+    props: Props,
+    init_label: string,
+    init_style: ReturnType<typeof style>
+  ) {
+    super(props);
+    this.state = {
+      state: 'Running',
+      label: init_label,
+      style: init_style,
+    };
+  }
+
+  render(): React.ReactElement {
+    return (
+      <div
+        className={classes(
+          'jp-ToolbarButton',
+          'jp-Toolbar-item',
+          toolbarItemClass
+        )}
+      >
+        <button
+          className={classes(
+            'bp3-button',
+            'bp3-minimal',
+            'jp-ToolbarButtonComponent',
+            'minimal',
+            'jp-Button',
+            toolbarButtonClass
+          )}
+          title={this.state.label}
+          onClick={this._onClick}
+        >
+          <span className={classes('bp3-button-text')}>
+            <span
+              className={classes(
+                'jp-ToolbarButtonComponent-icon',
+                toolbarButtonIconClass,
+                this.state.style
+              )}
+            >
+              {' '}
+            </span>
+          </span>
+        </button>
+      </div>
+    );
+  }
+
+  protected _onClick = (): void => {
+    console.log('hi');
```

Comment:
this can't merge to master

Reply:
this will be fixed in the next PR

Reply:
For things like this, I suggest you always put a comment 

# TODO(ashleyswang): placeholder for on click action, implementation is in the next PR.

When you work in a larger codebase, things like this will easily get forgotten, and rolled out to production.

Reply:
Also, for example, if you cannot finish implementing this in the internship, Lily will be able to search for all the TODOs in the codebase, instead of reading and understanding each file to find the missing pieces.